src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3829,74 +3825,102 @@ public Object other(DynamicObject string, int index) {
 
     @NonStandard
     @CoreMethod(names = "find_string_reverse", required = 2, lowerFixnum = 2)
+    @ImportStatic(StringGuards.class)
     public static abstract class StringRindexPrimitiveNode extends CoreMethodArrayArgumentsNode {
 
-        @Child private RopeNodes.GetByteNode patternGetByteNode = RopeNodes.GetByteNode.create();
-        @Child private RopeNodes.GetByteNode stringGetByteNode = RopeNodes.GetByteNode.create();
+        @Child private EncodingNodes.CheckEncodingNode checkEncodingNode;
 
-        @Specialization(guards = "isRubyString(pattern)")
-        public Object stringRindex(DynamicObject string, DynamicObject pattern, int start,
-                @Cached("create()") BranchProfile errorProfile,
-                @Cached("create()") RopeNodes.BytesNode stringBytes,
-                @Cached("create()") RopeNodes.BytesNode patternBytes) {
-            // Taken from Rubinius's String::rindex.
+        @Specialization(guards = "start < 0")
+        public Object stringRindexBadStart(DynamicObject string, DynamicObject pattern, int start) {
+            throw new RaiseException(getContext().getCoreExceptions().argumentError("negative start given", this));
+        }
 
-            int pos = start;
+        @Specialization(guards = { "start >= 0", "isEmpty(pattern)" })
+        public Object stringRindexEmptyPattern(DynamicObject string, DynamicObject pattern, int start) {
+            return start;
+        }
```

Comment:
Should this return the last index of the String?

Reply:
Not according to the original Rubinius implementation. And specs are all passing with it this way (it's also the way it worked in the old implementation).

Reply:
Weird. It doesn't make sense to me to return the first index if searching from the end.
Do the specs not pass if we return `string.length`?
I'd guess it's a Rubinius bug.